data/src/main/java/com/teamagam/gimelgimel/data/layers/LayersLocalCacheData.java

```diff
@@ -80,51 +76,19 @@ private URI downloadToCache(VectorLayer vectorLayer, URL url) {
     private File getVectorLayerFile(VectorLayer vectorLayer) {
         String fullFilename = mExternalVectorLayersDir +
                 File.separator +
-                generateTargetFilename(vectorLayer);
+                mLayerFilenameSerializer.toFilename(vectorLayer);
         return new File(fullFilename);
     }
 
-    private String generateTargetFilename(VectorLayer vectorLayer) {
-        ArrayList<String> nameElements = new ArrayList<>(4);
-        nameElements.add(LAYER_PREF, Constants.VECTOR_LAYER_CACHE_PREFIX);
-        nameElements.add(ID_POSITION, vectorLayer.getId());
-        nameElements.add(NAME_POSITION, vectorLayer.getName());
-        nameElements.add(VERSION_POSITION, String.valueOf(vectorLayer.getVersion()));
-        return TextUtils.join(NAME_SEPARATOR, nameElements) + KML_EXTENSION;
-    }
-
     private List<VectorLayer> extractVectorLayersFromFiles(File[] vectorLayerFiles) {
         List<VectorLayer> vectorLayers = new ArrayList<>(vectorLayerFiles.length);
         for (File file : vectorLayerFiles) {
             try {
-                vectorLayers.add(extractVectorLayerFromFile(file));
+                vectorLayers.add(mLayerFilenameSerializer.fromFilename(file.getName()));
```

Comment:
I concluded this after considering the extraction of a variable out of "mLayerFilenameSerializer.fromFilename(file.getName())"